spring-cloud-modules/spring-cloud-aws-v3/src/test/java/com/baeldung/spring/cloud/aws/sqs/SpringCloudAwsSQSLiveTest.java

```diff
@@ -0,0 +1,41 @@
+package com.baeldung.spring.cloud.aws.sqs;
+
+import io.awspring.cloud.sqs.operations.SqsTemplate;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.baeldung.spring.cloud.aws.sqs.BaeldungListeners.RECORD_PAYLOAD_QUEUE_NAME;
+import static com.baeldung.spring.cloud.aws.sqs.BaeldungListeners.STRING_PAYLOAD_QUEUE_NAME;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SpringCloudAwsSQSLiveTest extends BaseSqsIntegrationTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(SpringCloudAwsSQSLiveTest.class);
+
+    @Autowired
+    SqsTemplate sqsTemplate;
+
+    @Autowired
+    CountDownLatchContainer latchContainer;
+
+    @Test
+    void givenAStringPayload_whenSend_shouldReceive() throws Exception {
+        sqsTemplate.send(to -> to.queue(STRING_PAYLOAD_QUEUE_NAME).payload("My String"));
+        logger.info("Sent message");
+        assertThat(latchContainer.stringPayloadLatch.await(3, TimeUnit.SECONDS)).isTrue();
+    }
+
+    @Test
+    void givenARecordPayload_whenSend_shouldReceive() throws Exception {
+        sqsTemplate.send(to -> to
+                .queue(RECORD_PAYLOAD_QUEUE_NAME)
+                .payload(new BaeldungRecord(1, "My Record")));
+        logger.info("Sent message");
+        assertThat(latchContainer.recordPayloadLatch.await(3, TimeUnit.SECONDS)).isTrue();
+    }
```

Comment:
Currently, we are not proving that the expected message payload is received, only that "some message" is received, and doing so indirectly via the knowledge about what these latches are responsible for. 

Can we find a way to test the actual message payload too?

And can you think of a more direct way to test this scenario? I currently have to know about the relationship between CountDownLatchContainer and BaeldungListeners in order to understand what the test is proving. Ideally, it should be clear from looking at the test class alone what exactly is being tested.

Reply:
I added a note on the scenario in the JIRA ticket, please let me know what you think.

If we use a DB approach, we could definitely check what was persisted in the database.

Reply:
New PR supersedes this discussion I think